crates/biome_package/src/node_js_package/mod.rs

```diff
@@ -1,34 +1,46 @@
 mod package_json;
 mod tsconfig_json;
-
+use crate::{LICENSE_LIST, Manifest, Package, PackageAnalyzeResult, ProjectAnalyzeDiagnostic};
+use biome_parser::AnyParse;
+use biome_rowan::Language;
 pub use package_json::{Dependencies, PackageJson, PackageType, Version};
 pub use tsconfig_json::TsConfigJson;
 
-use biome_rowan::Language;
-
-use crate::{LICENSE_LIST, Manifest, Package, PackageAnalyzeResult, ProjectAnalyzeDiagnostic};
-
 #[derive(Default, Debug, Clone)]
 /// A Node.js project.
 pub struct NodeJsPackage {
     /// The `package.json` manifest
     pub manifest: Option<PackageJson>,
+    /// The raw manifest
+    pub raw_manifest: Option<AnyParse>,
     /// Diagnostics emitted during the operations
     pub diagnostics: Vec<biome_diagnostics::serde::Diagnostic>,
     /// The `tsconfig.json` manifest
-    pub tsconfig: TsConfigJson,
+    pub tsconfig: Option<TsConfigJson>,
 }
 
 impl NodeJsPackage {
     pub fn deserialize_tsconfig(&mut self, content: &ProjectLanguageRoot<TsConfigJson>) {
         let tsconfig = TsConfigJson::deserialize_manifest(content);
         let (tsconfig, deserialize_diagnostics) = tsconfig.consume();
-        self.tsconfig = tsconfig.unwrap_or_default();
+        self.tsconfig = tsconfig;
         self.diagnostics = deserialize_diagnostics
             .into_iter()
             .map(biome_diagnostics::serde::Diagnostic::new)
             .collect();
     }
+
+    pub fn to_deserialized_manifest(&self) -> Option<PackageJson> {
+        self.raw_manifest
+            .as_ref()
+            .and_then(|root| {
+                let node = root.tree();
+                let deserialized = <Self as Package>::Manifest::deserialize_manifest(&node);
+                let (manifest, _) = deserialized.consume();
+                manifest
+            })
+            .or_else(|| self.manifest.clone())
```

Comment:
I think we should try the deserialised manifest first, then fallback to the raw one. After that, we could insert the deserialised manifest, so we don't need to deserialise it again.

Reply:
We could, but unfortunately the architecture inside the workspace didn't allow me to use a mutable reference of `PackageJson`, that's why I used this approach. Not sure we can pin a list and mutate an item of the list, but if there's a solution please share it!

Reply:
Sorry, didn't see the immutability of &self. Then the `manifest` field is no longer needed?

Reply:
yes technically we don't need it anymore, however we might be able to optimise it and deserialise it only once and store the information somewhere. I already few ideas, but not required for this PR

Reply:
Yeah, agree. It's not blocker for this PR. Thank you for clarification!